### ISSUE TITLE
Connecting to MotherDuck with saas_mode=true returns Permission Error

### ISSUE THREAD

**Issue by matsonj** [duckdb/dbt-duckdb#426]
Consider the following dbt profile:

```
jdw:
  outputs:
    dev:
      type: duckdb
      schema: jdw
      path: 'md:jdw_dev?saas_mode=true'
      threads: 4
      extensions:
        - httpfs
  target: dev
  ```
  
When running `dbt build`, this error is returned to the terminal: `Permission Error: File system LocalFileSystem has been disabled by configuration`

dbt logs are below - believe it needs to auth to MotherDuck first and then read the catalog, not attempt to read the catalog from local first.

```
[0m12:55:54.564054 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'start', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x107226ba0>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x106ffb170>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x107275c10>]}


============================== 12:55:54.565821 | 965f4632-3809-4cdb-b962-a1c79e6389eb ==============================
[0m12:55:54.565821 [info ] [MainThread]: Running with dbt=1.8.5
[0m12:55:54.566120 [debug] [MainThread]: running dbt with arguments {'printer_width': '80', 'indirect_selection': 'eager', 'log_cache_events': 'False', 'write_json': 'True', 'partial_parse': 'True', 'cache_selected_only': 'False', 'profiles_dir': '/Users/jacobmatson/.dbt', 'fail_fast': 'False', 'debug': 'False', 'log_path': '/Users/jacobmatson/code/jdw/logs', 'warn_error': 'None', 'version_check': 'True', 'use_colors': 'True', 'use_experimental_parser': 'False', 'no_print': 'None', 'quiet': 'False', 'empty': 'False', 'log_format': 'default', 'invocation_command': 'dbt build --threads=1', 'warn_error_options': 'WarnErrorOptions(include=[], exclude=[])', 'introspect': 'True', 'target_path': 'None', 'static_parser': 'True', 'send_anonymous_usage_stats': 'True'}
[0m12:55:54.652719 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'project_id', 'label': '965f4632-3809-4cdb-b962-a1c79e6389eb', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x104f7c5c0>]}
[0m12:55:54.675700 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'adapter_info', 'label': '965f4632-3809-4cdb-b962-a1c79e6389eb', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x107317c80>]}
[0m12:55:54.676139 [info ] [MainThread]: Registered adapter: duckdb=1.8.2
[0m12:55:54.683166 [debug] [MainThread]: checksum: 643640a45b16e0d012867be4ac17daaa0c3ce97300faec3ce9a6de300b5e1c09, vars: {}, profile: , target: , version: 1.8.5
[0m12:55:54.722496 [debug] [MainThread]: Partial parsing enabled: 0 files deleted, 0 files added, 0 files changed.
[0m12:55:54.722710 [debug] [MainThread]: Partial parsing enabled, no changes found, skipping parsing
[0m12:55:54.735009 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'load_project', 'label': '965f4632-3809-4cdb-b962-a1c79e6389eb', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x110026810>]}
[0m12:55:54.772713 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'resource_counts', 'label': '965f4632-3809-4cdb-b962-a1c79e6389eb', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x110027080>]}
[0m12:55:54.772937 [info ] [MainThread]: Found 25 models, 24 sources, 410 macros
[0m12:55:54.774389 [info ] [MainThread]: 
[0m12:55:54.774644 [debug] [MainThread]: Acquiring new duckdb connection 'master'
[0m12:55:54.776749 [debug] [ThreadPool]: Acquiring new duckdb connection 'list_jdw_dev'
[0m12:55:54.797769 [debug] [ThreadPool]: Using duckdb connection "list_jdw_dev"
[0m12:55:54.797967 [debug] [ThreadPool]: On list_jdw_dev: /* {"app": "dbt", "dbt_version": "1.8.5", "profile_name": "jdw", "target_name": "dev", "connection_name": "list_jdw_dev"} */

        
    select schema_name
    from system.information_schema.schemata
    
    where catalog_name = '"jdw_dev"'
    
  
    
[0m12:55:54.798091 [debug] [ThreadPool]: Opening a new connection, currently in state init
[0m12:55:55.681082 [debug] [ThreadPool]: DuckDB adapter: Error running SQL: /* {"app": "dbt", "dbt_version": "1.8.5", "profile_name": "jdw", "target_name": "dev", "connection_name": "list_jdw_dev"} */

        
    select schema_name
    from system.information_schema.schemata
    
    where catalog_name = '"jdw_dev"'
    
  
    
[0m12:55:55.681726 [debug] [ThreadPool]: DuckDB adapter: Rolling back transaction.
[0m12:55:55.682787 [debug] [MainThread]: Connection 'master' was properly closed.
[0m12:55:55.683354 [debug] [MainThread]: Connection 'list_jdw_dev' was properly closed.
[0m12:55:55.683687 [info ] [MainThread]: 
[0m12:55:55.684026 [info ] [MainThread]: Finished running  in 0 hours 0 minutes and 0.91 seconds (0.91s).
[0m12:55:55.684447 [error] [MainThread]: Encountered an error:
Runtime Error
  Permission Error: File system LocalFileSystem has been disabled by configuration
[0m12:55:55.686160 [debug] [MainThread]: Resource report: {"command_name": "build", "command_wall_clock_time": 1.1479763, "process_user_time": 0.905537, "process_kernel_time": 0.136788, "process_mem_max_rss": "154533888", "command_success": false, "process_in_blocks": "0", "process_out_blocks": "0"}
[0m12:55:55.686713 [debug] [MainThread]: Command `dbt build` failed at 12:55:55.686622 after 1.15 seconds
[0m12:55:55.687035 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'end', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x107275c10>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10768b020>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x107638290>]}
[0m12:55:55.687338 [debug] [MainThread]: Flushing usage events
```

**Comment by guenp** [comment]
Thanks @matsonj for reporting! This looks like it's not a bug in `dbt-duckdb`, but I am investigating and will follow up here once we figure it out.

**Comment by guenp** [comment]
Hi @matsonj , just to loop back with you on this - it looks like a patch for a [security vulnerability](https://github.com/duckdb/duckdb/security/advisories/GHSA-w2gf-jxc9-pf2q) that caused this error. It will be fixed in DuckDB 1.1.0. I'm going to close this since it's not related to `dbt-duckdb`.
